optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -303,7 +304,11 @@ private void processSolutionCloner(DescriptorPolicy descriptorPolicy, PlanningSo
         if (solutionClonerClass != null) {
             solutionCloner = ConfigUtils.newInstance(this, "solutionClonerClass", solutionClonerClass);
         } else {
-            solutionCloner = new FieldAccessingSolutionCloner<>(this);
+            if (descriptorPolicy.getDomainAccessType() == DomainAccessType.GIZMO) {
+                solutionCloner = GizmoSolutionClonerFactory.build(this);
+            } else {
+                solutionCloner = new FieldAccessingSolutionCloner<>(this);
+            }
```

Comment:
code style: I 'd rather see a switch here so if we ever introduce a third DomainAccessType this code fails-fast if we forget to implement it here, instead of doing something silently under the covers. I guess the same applies for other usages of `descriptorPolicy.getDomainAccessType()` in previous PR's.

```
            switch (foo) {
                case AAA:
                    ...
                    break;
                case BBB:
                    ...;
                    break;
                default:
                    throw new IllegalStateException("The foo (" + foo + ") is not implemented.");
            }
```

What about null? Well, DescriptorPolicy.getDomainAccessType() never returns null and it's javadoc should state "@return never null".